### ISSUE TITLE
Fatal error when passing NULL to the Reader

### ISSUE THREAD

**Issue by alecpl** [sabre-io/vobject#657]
`Reader::read()`'s 1st argument is defined as `string|resource` indeed. However, it accepts NULL, and also `Parser/MimeDir::parse()`'s 1st argument is described as `string|resource|null`. So, there's some inconsistency. And passing NULL throws.
```
PHP Fatal error:  Uncaught TypeError: feof(): Argument #1 ($stream) must be of type resource, null given in /vendor/sabre/vobject/lib/Parser/MimeDir.php:283
Stack trace:
#0 /vendor/sabre/vobject/lib/Parser/MimeDir.php(283): feof(NULL)
#1 /vendor/sabre/vobject/lib/Parser/MimeDir.php(144): Sabre\VObject\Parser\MimeDir->readLine()
#2 /vendor/sabre/vobject/lib/Parser/MimeDir.php(90): Sabre\VObject\Parser\MimeDir->parseDocument()
#3 /vendor/sabre/vobject/lib/Reader.php(47): Sabre\VObject\Parser\MimeDir->parse(NULL, 3)
```

**Comment by phil-davis** [comment]
@alecpl see PR #658 

Do you think that this is enough - throwing an exception to avoid the PHP fatal error?

I could have tried to detect this directly in `Reader::read()` - but it seemed to work by detecting the problem inside `parse` and throwing the exception there.

**Comment by phil-davis** [comment]
Fixed by #658 
It will be in the next release "soon", and I will sort out back-porting the the various major release branches.

**Comment by phil-davis** [comment]
Released in https://github.com/sabre-io/vobject/releases/tag/4.5.5
